analysis/model/src/main/kotlin/de/maibornwolff/codecharta/model/Tree.kt

```diff
@@ -68,7 +71,8 @@ abstract class Tree<T> {
     fun getNodeBy(path: Path): Tree<T>? {
         return when {
             path.isTrivial -> this
-            path.isSingle  -> children.first { path == getPathOfChild(it) }
+            path.isSingle  ->
+                children.parallelStream().filter { path == getPathOfChild(it)}.collect(Collectors.toList()).first()
```

Comment:
This looks rather complicated. Since I do not know Kotlin that well, would you mind elaborating what this code does?